x-pack/plugin/downsample/src/main/java/org/elasticsearch/xpack/downsample/TransportDownsampleAction.java

```diff
@@ -897,6 +901,12 @@ private void createDownsampleIndex(
                 sourceIndexMetadata.getSettings().get(MapperService.INDEX_MAPPING_TOTAL_FIELDS_LIMIT_SETTING.getKey())
             );
         }
+        if (sourceIndexMetadata.getSettings().hasValue(FieldMapper.IGNORE_MALFORMED_SETTING.getKey())) {
+            builder.put(
+                FieldMapper.IGNORE_MALFORMED_SETTING.getKey(),
+                sourceIndexMetadata.getSettings().get(FieldMapper.IGNORE_MALFORMED_SETTING.getKey())
+            );
+        }
```

Comment:
I'm somewhat puzzled by this, so how does `@timestamp` pick up `ignore_above:true` if this is not set? I assume it's passed somewhere but not everywhere..

Reply:
I also don't fully understand this. Maybe the setting came from an index template?

Reply:
There are two sets of index settings:
1. Settings applied when index is created (what you are looking at).
2. Setting applied _after_ the index is created - everything from the original index is copied (except settings in 1).

I have to admit i don't understand the reason for the split but this is the code structure. `FieldMapper.IGNORE_MALFORMED_SETTING` was in 2 but now is in 1. The reason is that by the time 2 is applied it's too late to "fix" mapping of `@timestamp` with it in mind.